lib/chef/provider/user/useradd.rb

```diff
@@ -121,6 +121,8 @@ def universal_options
                 else
                   Chef::Log.debug("#{new_resource} setting home to #{new_resource.home}")
                   opts << "-d" << new_resource.home
+                  opts << "-M" if ['centos'].include?(node[:platform])
```

Comment:
the check needs to be:  `['rhel','fedora'].include?(node['platform_family'])`

otherwise you've only fixed it for centos and left everything else based on redhat broken.